devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,480 @@
+		"LED blink when Temperature report/Motion active": 2,
+		"LED blink when Motion active/LED off when Temperature report": 3]
+	parameter.parameter7default = "LED off when Temperature report/Motion active"
+	parameter.parameter7size = 1
+	
+	parameter.parameter8name = "RetryTimes"
+	parameter.parameter8title = "Motion notification retry times"
+	parameter.parameter8description = ""
+	parameter.parameter8type = "number"
+	parameter.parameter8options = "0..10"
+	parameter.parameter8enumMap = []
+	parameter.parameter8default = 3
+	parameter.parameter8size = 1
+	
+	return parameter."parameter${num}${text}"
+}
+
+def wakeUpIntervalUpdate(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" != value) {
+			state."${getWakeUpInfo("name")}" = value
+			state."${getWakeUpInfo("name")}Refresh" = true
+		}
+	}
+}
+
+def wakeUpIntervalCheck(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" == value) {
+			if (state."${getWakeUpInfo("name")}Refresh" == true) {		
+				state."${getWakeUpInfo("name")}Refresh" = false
+			}
+		}
+	}
+}
+
+def getWakeUpInfo(text) {
+	def wakeUp = [:]
+	
+	wakeUp.name = "wakeUpInterval"
+	wakeUp.title = "Wake up interval [Hours]"
+	wakeUp.description = ""
+	wakeUp.type = "number"
+	wakeUp.range = "1..4660"
+	wakeUp.default = 24
+	
+	return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
+	[
+		0x80: 1,
+		0x70: 2,
+		0X31: 5,
+		0x71: 3,
+		0x84: 2 
+	]
+}
+
+def parse(String description) {
+	def result = []
+	def cmd = zwave.parse(description, commandClassVersions)
+	
+	if (cmd) {
+		result += zwaveEvent(cmd)
+	} else {
+		logDebug "Unable to parse description: ${description}"
+	}
+	
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	
+	cmds += configure()
+	if (cmds) {
+		cmds << "delay 5000"
+	}
+	cmds << zwave.wakeUpV2.wakeUpNoMoreInformation().format()
+	
+	return cmds ? response(cmds) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	if (cmd.nodeid == zwaveHubNodeId) {
+		def interval = [:]
+		interval."${getWakeUpInfo("name")}" = cmd.seconds / 3600
+		wakeUpIntervalCheck(interval)
+	}
+	[]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%"]
+	
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "${device.displayName} has a low battery"
+		map.isStateChange = true
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	
+	state.lastBatteryReport = new Date().time
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def size = getConfigurationInfo(cmd.parameterNumber, "size")
+	def value = array2Integer(cmd.configurationValue)
```

Comment:
here and elsewhere we've provided as a convenience the `scaledConfigurationValue` where this conversion has already been done

conversely you can pass integers directly in configuration sets as well

you should be able to do away with your conversion functions

Reply:
Okay, I already use **scaledConfigurationValue** to replace my conversion functions. By the way, I can't see the **scaledConfigurationValue** in [class Configuration Report Command](https://graph.api.smartthings.com/ide/doc/zwave-utils.html#configurationV2), but **scaledConfigurationValue** still works fine in class Configuration Report Command.